src/dataflow/src/event.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Traits and types for describing captured timely dataflow streams.
+//!
+//! This is roughly based on [timely::dataflow::operators::event].
+
+use crate::activator::RcActivator;
+use std::borrow::Borrow;
+use timely::dataflow::operators::capture::event::EventIteratorCore;
+use timely::dataflow::operators::capture::{EventCore, EventPusherCore};
+
+/// An event link wrapper that activates targets on push.
```

Comment:
Naming nit for the future: nothing about this type requires this to be an `EventLink` inside.